Fix updates to globalSettings from Chrome Policy

jpevarnek opened the pull request:
Yeah, I'm sorry.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/uproxy/uproxy/2667)
<!-- Reviewable:end -->

trevj commented on src/generic_core/uproxy_core.ts:
```diff
@@ -777,11 +786,16 @@ export class uProxyCore implements uproxy_core_api.CoreApi {
-    this.updateGlobalSettings(globals.settings);
+  public updateOrgPolicy = (policy :uproxy_core_api.ManagedPolicyUpdate): void => {
+    // have to load settings first to make sure we don't overwrite anything
+    globals.loadSettings.then(() => {
```
why don't you use `updateGlobalSetting`?

trevj commented on src/generic_core/uproxy_core.ts:
```diff
@@ -777,11 +786,16 @@ export class uProxyCore implements uproxy_core_api.CoreApi {
+  public updateOrgPolicy = (policy :uproxy_core_api.ManagedPolicyUpdate): void => {
+    // have to load settings first to make sure we don't overwrite anything
+    globals.loadSettings.then(() => {
+      globals.settings.enforceProxyServerValidity = policy.
```
weird formatting

trevj commented on src/interfaces/uproxy_core_api.ts:
```diff
@@ -30,6 +30,12 @@ export enum UserFeedbackType {
   OTHER_FEEDBACK = 9
 }
 
+// Object containing an update to apply to another object
```
This is extremely abstract. Many other types here make it explicit for which operation they are, e.g. `CloudOperationArgs`. How about `UpdateGlobalSettingArgs`?

trevj commented on src/interfaces/uproxy_core_api.ts:
```diff
@@ -138,6 +144,7 @@ export enum Command {
   VERIFY_USER = 1033,
   VERIFY_USER_SAS = 1034,
   GET_PORT_CONTROL_SUPPORT = 1035,
+  UPDATE_GLOBAL_SETTING = 1036,
```
A comment here describing the payload would be great (though I see now literally no other value here specifies it, either).

trevj commented on src/interfaces/uproxy_core_api.ts:
```diff
@@ -173,7 +180,8 @@ export enum Update {
   // Payload is an integer between 0 and 100.
-  CLOUD_INSTALL_PROGRESS = 2029
+  CLOUD_INSTALL_PROGRESS = 2029,
+  REFRESH_GLOBAL_SETTINGS = 2030,
```
Again, a comment describing the payload.

trevj commented:
OK, I think I see what you're trying to do - although I'm lacking context on why this relates to Chrome Policy.

jpevarnek commented:
Eek, I forgot to put in a real PR description, sorry about that.

---

Review status: 0 of 17 files reviewed at latest revision, 5 unresolved discussions.

---

*[src/generic_core/uproxy_core.ts, line 791 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2667#-KOl7E3_bJVjJl7_0jQX-r1-791:-KOp61zw15bIERvQAnql:bf1249j) ([raw file](https://github.com/uproxy/uproxy/blob/7841d5a1d7d47406fea3d8379056f5d1dc66f001/src/generic_core/uproxy_core.ts#L791)):*
<details><summary><i>Previously, trevj wrote…</i></summary>
> why don't you use `updateGlobalSetting`?
</details>
It doesn't seem to make much sense to call the method that's going to end up taking these settings individually and then just set them with double calls to `storage.save` since we're already in the core.  I think you do make a very good point though that we should probably wait for `globals.loadSettings` in the core before setting anything.

---

*[src/generic_core/uproxy_core.ts, line 792 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2667#-KOl7E3_bJVjJl7_0jQX-r1-792:-KOp64M2_28KKXLxK9Eg:b41jnv4) ([raw file](https://github.com/uproxy/uproxy/blob/7841d5a1d7d47406fea3d8379056f5d1dc66f001/src/generic_core/uproxy_core.ts#L792)):*
<details><summary><i>Previously, trevj wrote…</i></summary>
> weird formatting
</details>
Eek.  Okay, I have no idea whatsoever how I even managed to type this.  Fixed.

---

*[src/interfaces/uproxy_core_api.ts, line 33 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2667#-KOl7E3_bJVjJl7_0jQk-r1-33:-KOp6d5NenUNCx0pF6vc:b-cfdz15) ([raw file](https://github.com/uproxy/uproxy/blob/7841d5a1d7d47406fea3d8379056f5d1dc66f001/src/interfaces/uproxy_core_api.ts#L33)):*
<details><summary><i>Previously, trevj wrote…</i></summary>
> This is extremely abstract. Many other types here make it explicit for which operation they are, e.g. `CloudOperationArgs`. How about `UpdateGlobalSettingArgs`?
</details>
Okay, fixed.

---

*[src/interfaces/uproxy_core_api.ts, line 147 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2667#-KOl7E3_bJVjJl7_0jQk-r1-147:-KOp76h33zRhBH9aCOLF:b-896fix) ([raw file](https://github.com/uproxy/uproxy/blob/7841d5a1d7d47406fea3d8379056f5d1dc66f001/src/interfaces/uproxy_core_api.ts#L147)):*
<details><summary><i>Previously, trevj wrote…</i></summary>
> A comment here describing the payload would be great (though I see now literally no other value here specifies it, either).
</details>
Done.

---

*[src/interfaces/uproxy_core_api.ts, line 184 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2667#-KOl7E3_bJVjJl7_0jQk-r1-184:-KOp774l4NV0mFiBuD3J:b-896fix) ([raw file](https://github.com/uproxy/uproxy/blob/7841d5a1d7d47406fea3d8379056f5d1dc66f001/src/interfaces/uproxy_core_api.ts#L184)):*
<details><summary><i>Previously, trevj wrote…</i></summary>
> Again, a comment describing the payload.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/uproxy/uproxy/2667#-:-KOp7Olbil5HFhkw1S9Y:bl7sk25)*
<!-- Sent from Reviewable.io -->

trevj commented:
:thumbsup:

---

Review status: 0 of 17 files reviewed at latest revision, 1 unresolved discussion.

---

*[src/generic_core/uproxy_core.ts, line 791 \[r1\]](https://reviewable.io:443/reviews/uproxy/uproxy/2667#-KOl7E3_bJVjJl7_0jQX-r1-791:-KOpu_lNGRskB_JxAzYY:b3gwf40) ([raw file](https://github.com/uproxy/uproxy/blob/7841d5a1d7d47406fea3d8379056f5d1dc66f001/src/generic_core/uproxy_core.ts#L791)):*
<details><summary><i>Previously, jpevarnek (Jonathan Pevarnek) wrote…</i></summary>
> It doesn't seem to make much sense to call the method that's going to end up taking these settings individually and then just set them with double calls to `storage.save` since we're already in the core.  I think you do make a very good point though that we should probably wait for `globals.loadSettings` in the core before setting anything.
</details>
Ah! Got it :-)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/uproxy/uproxy/2667#-:-KOpucXVzwe6A1HOCJul:bn1wz0t)*
<!-- Sent from Reviewable.io -->
